dynamiqs/utils/optimal_control.py

```diff
@@ -122,107 +91,34 @@ def rand_complex(
         ![rand_complex](/figs-code/rand_complex.png){.fig}
 
     Args:
-        size _(int or tuple of ints)_: Size of the returned tensor.
-        rmax: Maximum magnitude of the random complex numbers.
-        requires_grad: Whether gradients need to be computed with respect to the
-            returned tensor.
+        shape _(int or tuple of ints)_: Shape of the returned array.
+        rmax: Maximum magnitude.
         seed: Seed for the random number generator.
-        dtype: Complex data type of the returned tensor.
-        device: Device of the returned tensor.
+        dtype: Complex data type of the returned array.
 
     Returns:
-        _(*size)_ Tensor filled with random complex numbers.
+        _(array of shape (*shape))_ Array filled with random complex numbers.
 
     Examples:
-        >>> x = dq.rand_complex((2, 5), rmax=5.0, seed=42)
-        >>> x
-        tensor([[ 4.305-1.876j, -3.980-2.653j,  2.109-2.263j, -4.461-2.021j,
-                 -0.175-3.119j],
-                [-3.501+1.663j,  1.904-1.670j, -3.983-1.995j, -0.504+4.823j,
-                 -1.270-1.310j]])
+        >>> dq.rand_complex((2, 3), rmax=5.0, seed=42)
+        Array([[ 1.341+4.17j ,  3.978-0.979j, -2.592-0.946j],
+               [-4.428+1.744j, -0.53 +1.668j,  2.582+0.65j ]], dtype=complex64)
     """
-    # Note: We need to manually fetch the default device, because if `device` is `None`
-    # `torch.Generator` picks "cpu" as the default device, and not the device set by
-    # `torch.set_default_device`.
-    device = to_device(device)
-
-    # define random number generator from seed
-    generator = torch.Generator(device=device)
-    generator.seed() if seed is None else generator.manual_seed(seed)
-
-    rdtype = dtype_complex_to_real(get_cdtype(dtype))
-
-    rand = lambda: torch.rand(size, generator=generator, dtype=rdtype, device=device)
-
-    # generate random magnitude with values in [0, rmax[, the sqrt ensures that the
-    # resulting complex numbers are uniformly distributed in the complex plane
-    r = rmax * rand().sqrt()
-    # generate random phase with values in [0, 2pi[
-    theta = 2 * torch.pi * rand()
-    x = r * torch.exp(1j * theta)
-
-    # start tracking operations on the tensor if requires_grad is True
-    x.requires_grad_(requires_grad)
-
-    return x
-
-
-def pwc_pulse(times: Tensor, values: Tensor) -> callable[[float], Tensor]:
```

Comment:
Is this function not needed anymore?

Reply:
It's replaced by `dq.totime()` for pwc arrays.